Lib9c/Model/State/StakeState.cs

```diff
@@ -160,18 +160,90 @@ public int CalculateAccumulatedRuneRewards(long blockIndex, out int v1Step, out
             return v1Step + v2Step;
         }
 
+        /// <summary>
+        /// Calculate accumulated rewards step.
+        /// </summary>
+        /// <param name="currentBlockIndex">The block index of the current block.</param>
+        /// <param name="rewardStartBlockIndex">
+        /// The block index of the reward start block.
+        /// If not null, the return value is calculated differently like below.
+        /// # Case 1
+        ///   RewardInterval: 10, StartedBlockIndex: 0, ReceivedBlockIndex: 0,
+        ///   currentBlockIndex: 100, rewardStartBlockIndex: null
+        ///   -> 10
+        /// # Case 2
+        ///   RewardInterval: 10, StartedBlockIndex: 0, ReceivedBlockIndex: 15,
+        ///   currentBlockIndex: 100, rewardStartBlockIndex: null
+        ///   -> 9
+        /// # Case 3
+        ///   RewardInterval: 10, StartedBlockIndex: 0, ReceivedBlockIndex: 0,
+        ///   currentBlockIndex: 100, rewardStartBlockIndex: 15
+        ///   -> 8
+        /// # Case 4
+        ///   RewardInterval: 10, StartedBlockIndex: 0, ReceivedBlockIndex: 55,
+        ///   currentBlockIndex: 100, rewardStartBlockIndex: 15
+        ///   -> 5
+        /// # Case 5
+        ///   RewardInterval: 10, StartedBlockIndex: 0, ReceivedBlockIndex: 15,
+        ///   currentBlockIndex: 100, rewardStartBlockIndex: 55
+        ///   -> 4
+        /// # Case 6
+        ///   RewardInterval: 10, StartedBlockIndex: 100, ReceivedBlockIndex: 0,
+        ///   currentBlockIndex: 200, rewardStartBlockIndex: 55
+        ///   -> 10
+        /// # Case 7
+        ///   RewardInterval: 10, StartedBlockIndex: 100, ReceivedBlockIndex: 115,
+        ///   currentBlockIndex: 200, rewardStartBlockIndex: 55
+        ///   -> 9
+        /// </param>
+        /// <returns>The accumulated rewards step.</returns>
+        internal int GetRewardStep(long currentBlockIndex, long? rewardStartBlockIndex)
+        {
+            var stepBlockIndexes = new List<long>();
+            while (true)
+            {
+                var nextStepBlockIndex =
+                    StartedBlockIndex + RewardInterval * (stepBlockIndexes.Count + 1);
+                if (nextStepBlockIndex > currentBlockIndex)
+                {
+                    break;
+                }
+
+                stepBlockIndexes.Add(nextStepBlockIndex);
+            }
+
+            if (stepBlockIndexes.Count == 0)
+            {
+                return 0;
+            }
+
+            if (ReceivedBlockIndex > 0)
+            {
+                stepBlockIndexes = stepBlockIndexes
+                    .Where(stepBlockIndex => stepBlockIndex > ReceivedBlockIndex)
+                    .ToList();
+                if (stepBlockIndexes.Count == 0)
+                {
+                    return 0;
+                }
+            }
+
+            if (rewardStartBlockIndex.HasValue)
+            {
+                var validStepBlockIndex = rewardStartBlockIndex + RewardInterval;
```

Comment:
I wonder if the below case is intended. In other words, what means `rewardStartBlockIndex`?

```python
# Python
RewardInterval = 100
rewardStartBlockIndex = 250
StartedBlockIndex = 0
ReceivedBlockIndex = 100
currentBlockIndex = 300
stepBlockIndexes = []

# After loop
stepBlockIndexes = [100, 200, 300]

# After "ReceivedBlockIndex > 0" condition
stepBlockIndexes = [200, 300]

# After "rewardStartBlockIndex.HasValue" condition if statement
validStepBlockIndex = rewardStartBlockIndex + RewardInterval  # 350
stepBlockIndexes = [x for x in stepBlockIndexes if x >= validStepBlockIndex]. # []
```

Reply:
Confirmed by the game designer.